src/web/components/CMP.tsx

```diff
@@ -21,16 +17,25 @@ export const CMP = () => {
     const [show, setShow] = useState(false);
 
     useEffect(() => {
-        if (shouldShow()) {
-            setShow(true);
+        const { start, end } = initPerf(
+            'consent-management-platform-utilities',
+        );
+        start();
+        import(
+            /* webpackChunkName: "consent-management-platform-utilities" */ '@guardian/consent-management-platform'
+        ).then(({ shouldShow, setErrorHandler }) => {
+            end();
+            if (shouldShow()) {
+                setShow(true);
 
-            // setErrorHandler takes function to be called on errors in the CMP UI
-            setErrorHandler((errMsg: string): void => {
-                const err = new Error(errMsg);
+                // setErrorHandler takes function to be called on errors in the CMP UI
+                setErrorHandler((errMsg: string): void => {
+                    const err = new Error(errMsg);
 
-                window.guardian.modules.sentry.reportError(err, 'cmp');
-            });
-        }
+                    window.guardian.modules.sentry.reportError(err, 'cmp');
+                });
+            }
+        });
```

Comment:
Should this be inside a `useEffect` like this? I'm a little confused by what's gong on here. I though you can set the dynamic import at the top of the file and then it only gets pulled down when it's used. I'm not 100% following this though so I could be off

Reply:
The examples on webpack move the import logic within the function https://webpack.js.org/guides/code-splitting/#dynamic-imports

I think the behavior you are referring to is only accomplished thanks to `React.lazy`.